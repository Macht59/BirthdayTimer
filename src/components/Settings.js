
import React from "react";
import { AppRegistry, View, DatePickerIOS, Text, AsyncStorage, Button, Alert, DatePickerAndroid, Platform } from "react-native";
import { textStyles } from "../styles/textStyles";

export default class Settings extends React.Component {

    constructor(props) {
        super(props);

        this.setDate = this.setDate.bind(this);
        this.saveDate = this.saveDate.bind(this);
        this.openDataPicker = this.openDataPicker.bind(this);

        this.state = { chosenDate: new Date() };
    }

    render() {
        return (<View>
            <Text style={[textStyles.header, textStyles.shadow, { paddingTop: 50 }]}>Please, pick your birthdate</Text>
            {Platform.OS === "ios" && <DatePickerIOS
                date={this.state.chosenDate}
                onDateChange={this.setDate}
                mode="datetime"
            />}
            {Platform.OS === "ios" && <Button
                onPress={this.saveDate}
                title="Save my birthday!"
            />}
            {Platform.OS === "android" && <Text
                style={[textStyles.header, {

                }]}
            >{this.state.chosenDate.toLocaleDateString()}</Text>}
            {Platform.OS === "android" && <Button
                onPress={this.openDataPicker}
                title="Set date"
            />}
        </View>);
    }

    setDate(newDate) {
        this.setState({ chosenDate: newDate });
    }

    saveDate() {
        AsyncStorage.setItem("birthDate", this.state.chosenDate.getTime().toString());
        if (this.props.dateUpdated) {
            this.props.dateUpdated(this.state.chosenDate);
        }

        Alert.alert("Your birthdate has been saved!")
    }

    openDataPicker() {
        try {
            DatePickerAndroid.open({
                date: this.state.chosenDate
            }).then(value => {
                console.debug(value);
                if (value.action !== DatePickerAndroid.dismissedAction) {
                    const chosenDate = new Date(value.year, value.month, value.day);
                    this.setState({ chosenDate });
                    AsyncStorage.setItem("birthDate", this.state.chosenDate.getTime().toString());
                    if (this.props.dateUpdated) {
                        this.props.dateUpdated(this.state.chosenDate);
                    }
                }
            });
        } catch ({ code, message }) {
            console.warn('Cannot open date picker', message);
        }
    }
}

AppRegistry.registerComponent("Settings", () => Settings);
