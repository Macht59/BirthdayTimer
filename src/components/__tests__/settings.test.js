import React from 'react';
import Settings from '../settings';
import renderer from 'react-test-renderer';

test('Settings component renders correctly', () => {
    const birthDate = new Date(151515);
    const rendered = renderer.create(<Settings birthDate={birthDate} />).toJSON();
    expect(rendered).toMatchSnapshot();
});