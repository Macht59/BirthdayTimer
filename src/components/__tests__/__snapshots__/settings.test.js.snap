// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "skyblue",
          "fontSize": 30,
          "textAlign": "center",
        },
        Object {
          "textShadowColor": "rgba(255, 255, 255, 2)",
          "textShadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "textShadowRadius": 20,
        },
        Object {
          "paddingBottom": 25,
          "paddingTop": 25,
          "textAlign": "center",
        },
      ]
    }
  >
    Your birthday:
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <View>
      <RCTDatePicker
        date={151515}
        mode="date"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 216,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0984e3",
          "opacity": 0.9,
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontSize": 20,
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
      >
        Save my birthday!
      </Text>
    </View>
  </View>
</View>
`;
